server/bin/test_data_importer.py

```diff
@@ -0,0 +1,706 @@
+                })
+
+                # Future pool
+                pools.append({
+                    'product_id': product['id'],
+                    'quantity': large_quantity,
+                    'contract_number': 0,
+                    'account_number': account_number,
+                    'order_number': order_number,
+                    'start_date': (now + datetime.timedelta(days=30)).isoformat(),
+                    'end_date': end_date
+                })
+
+            else:
+                # Product is an engineering product. Flag the pid/owner combo so we can fetch the cert later
+                engineering_products[product['id']] = owner_key
+
+        log.info("  Creating %d subscription(s) for owner %s..." % (len(pools), owner_key))
+        for pool in pools:
+            cp.create_pool(owner_key, pool)
+
+        # Refresh (if necessary)
+        if cp.hosted_test:
+            log.info("  Refreshing owner: %s" % (owner_key))
+            cp.refresh_pools(owner_key)
+
+
+    # Create activation keys
+    for owner_key in owners:
+        pools = cp.list_pools(owner_key)
+        activation_keys.extend([(owner_key, "%s-%s-key-%s" % (owner_key, pool['productId'], pool['id']), pool) for pool in pools])
+
+    if activation_keys:
+        print('')
+        log.info("Creating %d activation keys..." % (len(activation_keys)))
+        for owner_key, key_name, pool in activation_keys:
+            log.info("  Activation key: %s [owner: %s, pool: %s]" % (key_name, owner_key, pool['id'] if pool is not None else 'n/a'))
+
+            key = cp.create_activation_key(owner_key, {'name': key_name})
+            if pool is not None:
+                cp.add_pool_to_activation_key(key['id'], pool['id'])
+
+
+    # Fetch entitlements/certs for engineering products
+    if engineering_products:
+        print('')
+        log.info("Fetching certs for %d engineering products..." % (len(engineering_products)))
+
+        # make sure the cert directory exists
+        if not os.path.exists(options.cert_dir):
+            os.makedirs(options.cert_dir)
+
+        for pid, owner_key in engineering_products.items():
+            try:
+                # This could 404 if the product did not get pulled down during refresh (hosted mode)
+                # because it's not linked/used by anything. We'll issue a warning in this case.
+                product_cert = cp.get_product_cert(owner_key, pid)
+                cert_filename = "%s/%s.pem" % (options.cert_dir, pid)
+
+                with open(cert_filename, 'w+') as file:
+                    log.info("  Writing certificate: %s" % (cert_filename))
+                    file.write(product_cert['cert'])
+            except requests.exceptions.HTTPError as e:
+                if e.response.status_code == 404:
+                    log.warn("  Skipping certificate for unused product: %s" % (pid))
+                else:
+                    raise e
+
+        print('')
```

Comment:
This `print()` and `log.info()` should be shifted left.

Reply:
Fixed